### PR TITLE
Cater for missing extensions in the restic backups.

### PR DESCRIPTION
**Pull request by seanhamlin**
# Checklist
- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [x] PR title is ready for changelog and subsystem label(s) applied

Explain the **details** for making this change. What existing problem does the pull request solve?

There was a bug in the k8up backup system that meant that the extension of the backups was missing in the webhook data. This PR adds more regex to cater for this.

# Closing issues

### CODE REVIEW

File: services/webhooks2tasks/src/handlers/resticbackupSnapshotFinished.js [hunk 1]
```diff
@@ -99,18 +99,41 @@ async function resticbackupSnapshotFinished(webhook: WebhookRequestData) {
...
+    // * drupal-example-site-master-solr-prebackuppod.solr.tar
+    // * drupal-example-site-master-solr-prebackuppod
+    // * drupal-example-site-master-mariadb-prebackuppod
+    // * drupal-example-site-master-mariadb-prebackuppod.mariadb.sql
+    //
+    // So we need to cater for these additional filenames.
     let source;
     const paths = R.prop('paths', snapshot);
-    const pattern = /^\/data\/([\w-]+)|([\w-]+).(?:sql|tar)$/;
+    const pattern = /^\/data\/([\w-]+)|([\w-]+)\.(?:sql|tar)$/;
```
Comment (Schnitzel) [inline services/webhooks2tasks/src/handlers/resticbackupSnapshotFinished.js]: couldn't we just add `/([\w]+)-prebackuppod/` as a third regex inside of here? not sure we really need the additional if check, but I let @rocketeerbkw decide this

Comment (seanhamlin) [reply services/webhooks2tasks/src/handlers/resticbackupSnapshotFinished.js]: 2 was hard enough to wrap my brain around ;) We could have a super regex

```
^\/data\/([\w-]+)|([\w-]+)\.(?:sql|tar)$|([\w]+)-prebackuppod
```

https://regex101.com/r/PNGPtd/5

Comment (Schnitzel) [reply services/webhooks2tasks/src/handlers/resticbackupSnapshotFinished.js]: yes! that one sounds good, can you implement that one?

Comment (seanhamlin) [reply services/webhooks2tasks/src/handlers/resticbackupSnapshotFinished.js]: Updated

**Comment by Schnitzel** [comment]
@rocketeerbkw can you have a look at this? I feel there would need to be an `R.prop(3, matches)` somewhere but you know R better than Sean and me

**Review by rocketeerbkw — CHANGES_REQUESTED** [review]
Ya, it probably needs another IF to check for the newly added regex group. Here's a [ramda REPL example](https://ramdajs.com/repl/#?function%20getSource%28path%29%20%7B%0Alet%20source%3B%0Aconst%20paths%20%3D%20%5Bpath%5D%3B%0A%0A%20%20const%20pattern%20%3D%20%2F%5E%5C%2Fdata%5C%2F%28%5B%5Cw-%5D%2B%29%7C%28%5B%5Cw-%5D%2B%29%5C.%28%3F%3Asql%7Ctar%29%24%7C%28%5B%5Cw%5D%2B%29-prebackuppod%2F%3B%0A%20%20if%20%28R.isEmpty%28paths%29%29%20%7B%0A%20%20%20%20source%20%3D%20%27unknown%27%3B%0A%20%20%7D%20else%20%7B%0A%20%20%20%20const%20path%20%3D%20R.head%28paths%29%3B%0A%20%20%20%20if%20%28%21R.test%28pattern%2C%20path%29%29%20%7B%0A%20%20%20%20%20%20source%20%3D%20%27unknown%27%3B%0A%20%20%20%20%7D%20else%20%7B%0A%20%20%20%20%20%20const%20matches%20%3D%20R.match%28pattern%2C%20path%29%3B%0A%20%20%20%20%20%20if%20%28R.prop%281%2C%20matches%29%29%20%7B%0A%20%20%20%20%20%20%20%20source%20%3D%20R.prop%281%2C%20matches%29%3B%0A%20%20%20%20%20%20%7D%20else%20if%20%28R.prop%282%2C%20matches%29%29%20%7B%0A%20%20%20%20%20%20%20%20source%20%3D%20R.prop%282%2C%20matches%29%3B%0A%20%20%20%20%20%20%7D%20else%20if%20%28R.prop%283%2C%20matches%29%29%20%7B%0A%20%20%20%20%20%20%20%20source%20%3D%20R.prop%283%2C%20matches%29%3B%0A%20%20%20%20%20%20%7D%20else%20%7B%0A%20%20%20%20%20%20%20%20source%20%3D%20%27unknown%27%3B%0A%20%20%20%20%20%20%7D%0A%20%20%20%20%7D%0A%20%20%7D%20%20%0A%20%20%0A%20%20return%20source%3B%0A%7D%0A%0Aconst%20tests%20%3D%20%5B%0A%20%20%27drupal-example-site-master-solr-prebackuppod.solr.tar%27%2C%0A%20%20%27drupal-example-site-master-solr-prebackuppod%27%2C%0A%20%20%27drupal-example-site-master-mariadb-prebackuppod%27%2C%0A%20%20%27drupal-example-site-master-mariadb-prebackuppod.mariadb.sql%27%2C%0A%20%20%27drupal-example-site-master-mariadb-prebackuppod.db.sql%27%2C%0A%20%20%27site-com-master-mariadb.mariadb.sql%27%2C%0A%20%20%27%2Fdata%2Fnginx%27%2C%0A%20%20%27%2Fdata%2Fbanana%27%2C%0A%5D%3B%0A%0Aconst%20results%20%3D%20tests.map%28test%20%3D%3E%20getSource%28test%29%29%3B%0Aconsole.log%28tests%29%3B%0Aconsole.log%28results%29%3B) with the added check. This can probably be done in a less obtuse way... since the "super" regex is just 3 ORs that don't depend on eachother it could be split into 3 different regex and 3 diff if statements. Or a switch statement... or if this small change just works then `/shrug`

It looks like it's returning all the correct info? Not sure if we should be checking for the prebackuppod match BEFORE the *.tar one though? @seanhamlin you're more familiar with the bug, I'll leave that up to you.

**Comment by Schnitzel** [comment]
@rocketeerbkw 
the Ramda example is correct!
I switched the order and it doesn't make any difference, so I think this is fine.
@seanhamlin can you implement the changes so it fits the example?

**Comment by Schnitzel** [comment]
actually I had time and I added it right away :)

**Pull request merged**